### ISSUE TITLE
ButtonHelper MovieClip.hitTest() doesn't work in certain situations, can fail to hit

### ISSUE THREAD

**Issue by pittins** [CreateJS/EaselJS#635]
tl;dr: A bit of an unusual situation, but MovieClip.hitTest() fails unless MovieClip._updateTimeline() is called first (it is most usually called indirectly when stage.enableMouseOver() is on as part of its own hit testing).

Flash Pro generates ButtonHelper instances to help with managing button states. Theoretically, all you have to do is stage.enableMouseOver() and set a frequently in order to have the button's states change to hover, pressed, and "up" automatically.

However, if you try and hitTest the button's hitArea without previously calling stage.enableMouseOver() (or before it executes for the first time, or if you are rolling some custom hovering code), you'll never get a hit as the hitTest will always fail.

The root issue, is that since the hitArea is not actually a part of the stage, it never gets its timeline initialized, and its tweens are never checked. Several functions initialize the timeline, including the enableMouseOver() callback and MovieClip.draw(). They all do this by (usually indirectly) calling MovieClip._updateTimeline().

Here's an example. The following is created by Flash Pro's output. It creates a button instance, gives it a new ButtonHelper, and tweens the button instance. Two instances of "button_yes" are created. One instance is the actual button (this.yesButton). The second instance is the "button helper" instance. This second instance is also set to the button's hitArea property, as part of the ButtonHelper()'s constructor. All good so far.

```javascript
// yesButton
this.yesButton = new lib.button_yes();
this.yesButton.setTransform(505.5,901.8,0.01,0.01);
this.yesButton._off = true;
new cjs.ButtonHelper(this.yesButton, 0, 1, 2, false, new lib.button_yes(), 3);
this.timeline.addTween(cjs.Tween.get(this.yesButton)....[TWEEN DATA HERE]);
```

Now if you just try and call yesButton.hitArea.hitTest() with some coordinates, it will always fail. That is because the hitArea is in a bit of an invalid state: It actually has no children attached, and has NaN as its currentFrame and _prevPos properties. Simply calling _updateTimeline() on it will cause its state to correct itself: It will go to frame 3 and have Shapes added as its children.

The simplest solution to this, would be to just add an override MovieClip.hitTest() that has the same one line of code as Container.hitTest() (or supers into it), but also calls MovieClip._updateTimeline() first to ensure that the MovieClip instance is in a proper state:

```javascript
// in MovieClip's prototype list
p.hitTest = function(x, y) {
  this._updateTimeline();
  this.Container_hitTest(x, y);
};
```

I feel like this is a proper solution, because it ensures that no matter the state of the stage's mouse over capabilities, a hitTest will yield the proper result. Testing indicates that this works. My only concern is that frame functions might be called when they are not supposed to, as calls to the frame functions are themselves tweened. You guys might have a better suggestion.
